Use hidden symbol visibility for Abseil's C++ code.

This probably improves linker and loader throughput.

Reasoning for not upstreaming this patch: would need too much surgery to
gold-plate it for all of their supported build modes.

--- absl/base/config.h.orig
+++ absl/base/config.h
@@ -153,7 +153,7 @@
 #define ABSL_INTERNAL_C_SYMBOL(x) x
 #elif ABSL_OPTION_USE_INLINE_NAMESPACE == 1
 #define ABSL_NAMESPACE_BEGIN \
-  inline namespace ABSL_OPTION_INLINE_NAMESPACE_NAME {
+  inline namespace ABSL_OPTION_INLINE_NAMESPACE_NAME __attribute__ ((visibility ("hidden"))) {
 #define ABSL_NAMESPACE_END }
 #define ABSL_INTERNAL_C_SYMBOL_HELPER_2(x, v) x##_##v
 #define ABSL_INTERNAL_C_SYMBOL_HELPER_1(x, v) \
--- absl/synchronization/mutex.cc.orig
+++ absl/synchronization/mutex.cc
@@ -2784,7 +2784,7 @@
 }
 
 #ifdef ABSL_HAVE_THREAD_SANITIZER
-extern "C" void __tsan_read1(void* addr);
+extern "C" void __tsan_read1(void* addr) __attribute__((visibility ("default")));
 #else
 #define __tsan_read1(addr)  // do nothing if TSan not enabled
 #endif
